ProductComment.php

```diff
@@ -1,443 +1,459 @@
+			FROM `' . _DB_PREFIX_ . 'product_comment` pc
+			WHERE pc.`id_product` = ' . (int) $id_product . '
+			AND pc.`deleted` = 0' .
+            ($validate == '1' ? ' AND pc.`validate` = 1' : '');
 
+        return Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow($sql);
+    }
 
-		return Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow($sql);
+    public static function getAverageGrade($id_product)
+    {
+        $validate = Configuration::get('PRODUCT_COMMENTS_MODERATE');
 
-	}
-
-	public static function getAverageGrade($id_product)
-	{
-		$validate = Configuration::get('PRODUCT_COMMENTS_MODERATE');
-
-		return Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow('
+        return Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow('
 		SELECT (SUM(pc.`grade`) / COUNT(pc.`grade`)) AS grade
-		FROM `'._DB_PREFIX_.'product_comment` pc
-		WHERE pc.`id_product` = '.(int)$id_product.'
-		AND pc.`deleted` = 0'.
-		($validate == '1' ? ' AND pc.`validate` = 1' : ''));
-	}
-
-	public static function getAveragesByProduct($id_product, $id_lang)
-	{
-		/* Get all grades */
-		$grades = ProductComment::getGradeByProduct((int)$id_product, (int)$id_lang);
-		$total = ProductComment::getGradedCommentNumber((int)$id_product);
-		if (!count($grades) || (!$total))
-			return array();
-
-		/* Addition grades for each criterion */
-		$criterionsGradeTotal = array();
-		$count_grades = count($grades);
-		for ($i = 0; $i < $count_grades; ++$i)
-			if (array_key_exists($grades[$i]['id_product_comment_criterion'], $criterionsGradeTotal) === false)
-				$criterionsGradeTotal[$grades[$i]['id_product_comment_criterion']] = (int)($grades[$i]['grade']);
-			else
-				$criterionsGradeTotal[$grades[$i]['id_product_comment_criterion']] += (int)($grades[$i]['grade']);
-
-		/* Finally compute the averages */
-		$averages = array();
-		foreach ($criterionsGradeTotal as $key => $criterionGradeTotal)
-			$averages[(int)($key)] = (int)($total) ? ((int)($criterionGradeTotal) / (int)($total)) : 0;
-		return $averages;
-	}
-
-	/**
-	 * Return number of comments and average grade by products
-	 *
-	 * @return array Info
-	 */
-	public static function getCommentNumber($id_product)
-	{
-		if (!Validate::isUnsignedId($id_product))
-			return false;
-		$validate = (int)Configuration::get('PRODUCT_COMMENTS_MODERATE');
-		$cache_id = 'ProductComment::getCommentNumber_'.(int)$id_product.'-'.$validate;
-		if (!Cache::isStored($cache_id))
-		{
-			$result = (int)Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
+		FROM `' . _DB_PREFIX_ . 'product_comment` pc
+		WHERE pc.`id_product` = ' . (int) $id_product . '
+		AND pc.`deleted` = 0' .
+        ($validate == '1' ? ' AND pc.`validate` = 1' : ''));
+    }
+
+    public static function getAveragesByProduct($id_product, $id_lang)
+    {
+        /* Get all grades */
+        $grades = ProductComment::getGradeByProduct((int) $id_product, (int) $id_lang);
+        $total = ProductComment::getGradedCommentNumber((int) $id_product);
+        if (!count($grades) || (!$total)) {
```

Comment:
ProductComment::getGradeByProduct() can return boolean, maybe use empty() instead of count()

Reply:
This actually the old code, it is not used any more but I won't delete it before the back office is also migrated. No need to improve this code.